packages/app/src/sections/futures/Trade/TransferSmartMarginModal.tsx

```diff
@@ -0,0 +1,271 @@
+	const [amount, setAmount] = useState('')
+	const [transferType, setTransferType] = useState(defaultTab === 'deposit' ? 0 : 1)
+
+	const susdBal = transferType === 0 ? susdBalance : totalWithdrawable
+
+	const balanceStatus: BalanceStatus = useMemo(
+		() =>
+			availableMargin.gt(ZERO_WEI) || susdBalance.gt(minDeposit)
+				? 'high_balance'
+				: susdBalance.eq(ZERO_WEI)
+				? 'no_balance'
+				: 'low_balance',
+		[availableMargin, minDeposit, susdBalance]
+	)
+
+	useEffect(() => {
+		switch (balanceStatus) {
+			case 'no_balance':
+			case 'low_balance':
+				setOpenSocket(true)
+				break
+			case 'high_balance':
+				setOpenSocket(false)
+		}
+	}, [balanceStatus])
+
+	const isDisabled = useMemo(() => {
+		if (!amount || submitting) {
+			return true
+		}
+		const amtWei = wei(amount)
+		if (amtWei.eq(0) || amtWei.gt(susdBal) || (transferType === 0 && amtWei.lt(minDeposit))) {
+			return true
+		}
+		return false
+	}, [amount, susdBal, minDeposit, transferType, submitting])
+
+	const computedWithdrawAmount = useMemo(
+		() => (availableMargin.eq(wei(amount || 0)) ? availableMargin : wei(amount || 0)),
+		[amount, availableMargin]
+	)
+
+	const onChangeShowSocket = useCallback(() => setOpenSocket(!openSocket), [openSocket])
+
+	const handleSetMax = useCallback(() => {
+		if (transferType === 0) {
+			setAmount(susdBal.toString())
+		} else {
+			setAmount(totalWithdrawable.toString())
+		}
+	}, [transferType, susdBal, totalWithdrawable])
+
+	const onChangeTab = (selection: number) => {
+		setTransferType(selection)
+		setAmount('')
+	}
+
+	const onDeposit = () => {
+		dispatch(depositCrossMargin(wei(amount)))
+	}
+
+	const onWithdraw = () => {
+		dispatch(withdrawCrossMargin(computedWithdrawAmount))
+	}
+
+	return (
+		<StyledBaseModal
+			title={
+				transferType === 0
+					? t('futures.market.trade.margin.modal.deposit.title')
+					: t('futures.market.trade.margin.modal.withdraw.title')
+			}
+			isOpen
+			onDismiss={onDismiss}
+		>
+			{balanceStatus === 'high_balance' ? (
+				<StyledSegmentedControl
+					values={['Deposit', 'Withdraw']}
+					selectedIndex={transferType}
+					onChange={onChangeTab}
+				/>
+			) : balanceStatus === 'no_balance' ? (
+				<Disclaimer>{t('futures.market.trade.margin.modal.bridge.no-balance')}</Disclaimer>
+			) : (
+				<Disclaimer>{t('futures.market.trade.margin.modal.bridge.low-balance')}</Disclaimer>
+			)}
+			{transferType === 0 && (
+				<>
+					<StyledCardHeader onClick={onChangeShowSocket} noBorder={openSocket}>
+						<BalanceText>{t('futures.market.trade.margin.modal.bridge.title')}</BalanceText>
+						{openSocket ? <CaretUpIcon /> : <CaretDownIcon />}
+					</StyledCardHeader>
+					{openSocket ? <SocketBridge /> : <Spacer height={20} />}
+				</>
+			)}
+			<BalanceContainer>
+				<BalanceText>{t('futures.market.trade.margin.modal.balance')}:</BalanceText>
+				<BalanceText>
+					<span>{formatDollars(susdBal)}</span> sUSD
+				</BalanceText>
+			</BalanceContainer>
+			<NumericInput
```

Comment:
We should only display this on withdraw and just show socket on deposit.

Reply:
True. Updated.